pkg/inject/virtualgateway_envoy.go

```diff
@@ -128,6 +128,9 @@ func (m *virtualGatewayEnvoyConfig) buildTemplateVariables(pod *corev1.Pod) Virt
 	meshName := m.getAugmentedMeshName()
 	virtualGatewayName := aws.StringValue(m.vg.Spec.AWSName)
 	preview := m.getPreview(pod)
+	if m.mutatorConfig.enableSDS && isSDSDisabled(pod) {
+		m.mutatorConfig.enableSDS = false
```

Comment:
This reads a bit off changing state of `m.mutatorConfig.enableSDS` for the entire package. i.e. it leaks the bounds of this specific function. It is better to use separate function local variable that is passed as a value of `EnableSDS ` in `VirtualGatewayEnvoyVariables`

Same applies to the change in envoy.go

Reply:
This setting is per Pod (i.e.,) inject config is populated with defaults per pod and we want to toggle the SDS support for Pod level anyways.